src/host/posix/cli_daemon.cpp

```diff
@@ -47,67 +47,199 @@
 
 namespace otbr {
 
+otbrError CliDaemon::Dependencies::InputCommandLine(const uint8_t *aBuf, uint16_t aLength)
+{
+    OTBR_UNUSED_VARIABLE(aBuf);
+    OTBR_UNUSED_VARIABLE(aLength);
+
+    return OTBR_ERROR_NONE;
+}
+
 static constexpr char kDefaultNetIfName[] = "wpan0";
 static constexpr char kSocketBaseName[]   = "/run/openthread-";
 static constexpr char kSocketSuffix[]     = ".sock";
 static constexpr char kSocketLockSuffix[] = ".lock";
 
 static constexpr size_t kMaxSocketFilenameLength = sizeof(sockaddr_un::sun_path) - 1;
 
-std::string CliDaemon::GetSocketFilename(const char *aSuffix) const
+std::string CliDaemon::GetSocketFilename(const std::string &aNetIfName, const char *aSuffix) const
 {
     std::string fileName;
-    std::string netIfName = mNetifName.empty() ? kDefaultNetIfName : mNetifName;
+
+    std::string netIfName = aNetIfName.empty() ? kDefaultNetIfName : aNetIfName;
 
     fileName = kSocketBaseName + netIfName + aSuffix;
     VerifyOrDie(fileName.size() <= kMaxSocketFilenameLength, otbrErrorString(OTBR_ERROR_INVALID_ARGS));
 
     return fileName;
 }
 
-CliDaemon::CliDaemon(void)
+CliDaemon::CliDaemon(Dependencies &aDependencies)
     : mListenSocket(-1)
     , mDaemonLock(-1)
+    , mSessionSocket(-1)
+    , mDeps(aDependencies)
 {
 }
 
-void CliDaemon::CreateListenSocketOrDie(void)
+otbrError CliDaemon::CreateListenSocket(const std::string &aNetIfName)
 {
+    otbrError error = OTBR_ERROR_NONE;
+
+    std::string        lockfile;
+    std::string        socketfile;
     struct sockaddr_un sockname;
 
     mListenSocket = SocketWithCloseExec(AF_UNIX, SOCK_STREAM, 0, kSocketNonBlock);
-    VerifyOrDie(mListenSocket != -1, strerror(errno));
+    VerifyOrExit(mListenSocket != -1, error = OTBR_ERROR_ERRNO);
 
-    std::string lockfile = GetSocketFilename(kSocketLockSuffix);
-    mDaemonLock          = open(lockfile.c_str(), O_CREAT | O_RDONLY | O_CLOEXEC, 0600);
-    VerifyOrDie(mDaemonLock != -1, strerror(errno));
+    lockfile    = GetSocketFilename(aNetIfName, kSocketLockSuffix);
+    mDaemonLock = open(lockfile.c_str(), O_CREAT | O_RDONLY | O_CLOEXEC, 0600);
+    VerifyOrExit(mDaemonLock != -1, error = OTBR_ERROR_ERRNO);
 
-    VerifyOrDie(flock(mDaemonLock, LOCK_EX | LOCK_NB) != -1, strerror(errno));
+    VerifyOrExit(flock(mDaemonLock, LOCK_EX | LOCK_NB) != -1, error = OTBR_ERROR_ERRNO);
 
-    std::string socketfile = GetSocketFilename(kSocketSuffix);
+    socketfile = GetSocketFilename(aNetIfName, kSocketSuffix);
     memset(&sockname, 0, sizeof(struct sockaddr_un));
```

Comment:
I feel it more clear to move the `memset` of sockname to the same block as filling the sockname.